src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/config/OAuth2ResourceServerSecurityConfiguration.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.core.convert.converter.Converter
+import org.springframework.http.HttpMethod
+import org.springframework.security.authentication.AbstractAuthenticationToken
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
+import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter
+import org.springframework.security.web.SecurityFilterChain
+
+@EnableWebSecurity
+class OAuth2ResourceServerSecurityConfiguration {
+  @Bean
+  @Throws(Exception::class)
+  fun securityFilterChain(http: HttpSecurity, @Autowired objectMapper: ObjectMapper): SecurityFilterChain {
+    http
+      .authorizeHttpRequests { authorize ->
+        authorize
+          .mvcMatchers(HttpMethod.GET, "/premises").authenticated()
```

Comment:
We tend to come at this from the opposite end - list the anonymous endpoints such as /health and open api docs, then require everything else to be authenticated. [Random example](https://github.com/ministryofjustice/create-and-vary-a-licence-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/ResourceServerConfiguration.kt#L15).

Reply:
So one advantage of doing it this way around is that it becomes impossible for endpoints that require a higher level of authorisation to slip through PR without an entry actually being added here.  e.g. if just having a JWT is sufficient for most endpoints on the service but for a new POST operaiton we also require an authority it's fairly easily done for someone to forget to add the requisite entry here.  Ideally this would be caught by testing + PR review but it is an easy one to miss.  That said, I don't know if that's actually going to be the case for this project yet so happy to switch for consistency sake if you'd like?

Also seems I forgot to add the entries for the healthcheck, swagger etc. so will pop those in.

Reply:
Ah ok, so does that mean you would expect to add the authorities here? We tend to use the `@PreAuthorize` annotation for that which is nice because then the required authority is next to the endpoint it secures. And we'd generally write an integration test for the required authority before adding it to the endpoint making it harder to miss. (You'll almost certainly end up securing your endpoints with a role btw. that's a fairly standard pattern we use).

All that said I don't think there's anything wrong with your approach. And consistency is a good thing generally, but herd mentality isn't 😁

Reply:
```
Ah ok, so does that mean you would expect to add the authorities here?
```

We've gone for the approach of using [Open Api Generator](https://github.com/OpenAPITools/openapi-generator/) to create the request/response models and controllers from the OpenAPI spec file so there's no chance of the spec being inconsistent with the actual implementation ([Pull Request here](https://github.com/ministryofjustice/approved-premises-api/pull/14) if interested.)  One downside of that is that we can't easily add @PreAuthorize to the controllers since they're generated so I would expect to add the authorities here.

Reply:
Ah interesting, contract first. I tried that generator a few years and it worked ok with some small compromises. Cool makes sense to me.